src/Features/Lsif/Generator/Generator.cs

```diff
@@ -64,14 +67,38 @@ private Generator(ILsifJsonWriter lsifJsonWriter)
         public static Generator CreateAndWriteCapabilitiesVertex(ILsifJsonWriter lsifJsonWriter)
         {
             var generator = new Generator(lsifJsonWriter);
-            var capabilitiesVertex = new Capabilities(generator._idFactory,
-                HoverProvider, DeclarationProvider, DefinitionProvider, ReferencesProvider,
-                TypeDefinitionProvider, DocumentSymbolProvider, FoldingRangeProvider, DiagnosticProvider);
+
+            // Pass the set of supported SemanticTokenTypes. Order must match
+            // the order used for serialization of semantic tokens array. This
+            // array is analogous to the equivelent array in https://microsoft.github.io/language-server-protocol/specifications/lsp/3.18/specification/#textDocument_semanticTokens.
+            //
+            // Ideally semantic tokens support would use the minimal set of token
+            // types specified in LSP's SemanticTokenTypes to reduce the number
+            // of tokens a particular consumer must understand, but Roslyn currently
+            // employs a large number of custom token types that aren't well supported
+            // by this existing set so we pass both LSP and Roslyn custom token types.
```

Comment:
I updated the comment. Effectively, there is a standardized set of semantic tokens that we would prefer the languages to use because they will be guaranteed to be themed by whomever consumes the LSIF. This set is missing certain Roslyn specific tokens, however, so I am including your custom token types.